pkg/client-sdk/example/main.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"sync"
+
+	arksdk "github.com/ark-network/ark-sdk"
+	log "github.com/sirupsen/logrus"
+)
+
+func main() {
+	var (
+		explorerUrl = "http://localhost:3001"
+		network     = "regtest"
+		aspUrl      = "http://localhost:6000"
+		aspPubKey   = "ASP_PUBKEY"
+
+		ctx         = context.Background()
+		explorerSvc = arksdk.NewExplorer(explorerUrl)
+	)
+
+	alice, err := arksdk.NewSingleKeyWallet(
+		explorerSvc, network, "",
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
would be ideal to instantiate this into the factory function below `arkdsk.New()` by providing a `walletType="singleKey"` or similar param

Reply:
Check https://github.com/ark-network/ark/pull/193#issuecomment-2217200874 comment, i think it would be ideal if we come up with good wallet abstraction and pass it as interface than to "leave" sdk to do creation of concrete wallets

Reply:
I see the benefit of decoupling wallet to be standalone packages, but in the end we are worsening the Dev UX, as it will result in more lines of code.

In this specific I am not sure we should call it as SingleKey, rather we should identify it based on the secret to be passed:

- PrivateKey 
- Mnemonic 


The fact in our implementation we decide to do public key reuse doesn't matter if the SDK is loaded with a mnemonic for example.